fdadb/management/commands/fetch_ndc_database.py

```diff
@@ -59,21 +66,50 @@ def handle(self, *args, **options):
         # * SUBSTANCENAME - names of the substances in the drug, e.g.: CLINDAMYCIN PHOSPHATE; TRETINOIN
         # * ACTIVE_NUMERATOR_STRENGTH - strengths of the substances, e.g.: 12; .25
         # * ACTIVE_INGRED_UNIT - units for the strengths e.g.: mg/g; mg/g
+
+        cache = {}
+
+        def cached_get_or_create(key, model, defaults=None, **kwargs):
+            try:
+                return cache[key], False
+            except KeyError:
+                obj, created = model.objects.get_or_create(defaults=defaults, **kwargs)
+                cache[key] = obj
+                return obj, created
+
         products_data = self.get_products_data()
-        for product in products_data:
-            print("Parsing: ", product["PROPRIETARYNAME"], product["PRODUCTNDC"])
+        skipped = 0
+        for i, product in enumerate(products_data):
+            if (i % 100) == 0:
+                print("\rrow %s" % i, end="", flush=True)
             if product["NDC_EXCLUDE_FLAG"] != "N":
-                print("Skipping excluded drug:", product["PROPRIETARYNAME"], product["PRODUCTNDC"])
+                skipped += 1
                 continue
             product_name = "{} {}".format(
-                product["PROPRIETARYNAME"].strip(), product["PROPRIETARYNAMESUFFIX"].strip()).strip()
+                product["PROPRIETARYNAME"].strip(),
+                product["PROPRIETARYNAMESUFFIX"].strip(),
+            ).strip()
             strength_data = self.get_medication_strength_data(product)
-            medication_name, created = MedicationName.objects.get_or_create(name=product_name, defaults={
-                "active_substances": list(strength_data.keys())
-            })
-            medication_strength, created = MedicationStrength.objects.get_or_create(
-                medication_name=medication_name, strength=strength_data)
-            MedicationNDC.objects.get_or_create(ndc=product["PRODUCTNDC"], defaults={
-                "medication_strength": medication_strength,
-                "manufacturer": product["LABELERNAME"]
-            })
+            medication_name, created = cached_get_or_create(
+                "name:%s" % (product_name,),
```

Comment:
use python str.format - in general, it's recommended by the Python community as more safe than the old printf style string format

Reply:
Good catch.
(I've always disliked `.format` and tend to avoid it in personal projects, but yeah, it's more pythonic for the phase between `%` and `f""`)